pkgs/build-support/build-fhsenv-bubblewrap/default.nix

```diff
@@ -27,13 +27,13 @@
 , ...
 } @ args:
 
-assert (pname != null || version != null) -> (name == null && pname != null); # You must declare either a name or pname + version (preferred).
+assert (pname == null || version == null) -> (name != null); # You must declare name if pname + version (preferred) is missing.
 
 with builtins;
 let
-  pname = if args ? name && args.name != null then args.name else args.pname;
-  versionStr = lib.optionalString (version != null) ("-" + version);
-  name = pname + versionStr;
+  name = if args ? name && args.name != null then args.name else "${pname}-${version}";
+  pname = if args ? pname && args.pname != null then args.pname else lib.getName args.name;
+  version = if args ? version && args.version != null then args.version else lib.getVersion args.name;
```

Comment:
As mentioned in https://github.com/NixOS/nixpkgs/pull/262506, I don't think we should ever "infer" pname from a name when we could instead pass pname directly.

Reply:
I kinda agree, but would alternative be to make pname required, or to fallback to name? That issue seems intent on not affecting existing packages while i figured there aren't that many to fix yet

Reply:
bump. Personally i'd like to decouple the basename of the resulting `$out/bin` binary produced in `buildFHSEnvBubblewrap` from `pname` and `name` entirely.

Reply:
ftr, all the packages i fix in this PR now passes in `pname` explicitly.

Reply:
That sounds like a good idea and I'm for an additional `binaryName` arg that defaults to `pname` but it has nothing to do with the concern I voiced here.

Reply:
> but it has nothing to do with *the concern I voiced here*.

Then _please_ reply to our questions about what it _is_ you prefer. Bith i ask clarifying questions (https://github.com/NixOS/nixpkgs/pull/271071#discussion_r1412737161) abd the parallel effort asks clarifying questions (https://github.com/NixOS/nixpkgs/pull/262506#issuecomment-1817481064). The only feedback I'm working with is:

* > I've tried adding pname support to appimage tools "properly" a little while ago but haven't found the time to finish it yet.
* > Please split this out into two PRs.
* > I don't think we should ever "infer" pname from a name

Reply:
https://github.com/NixOS/nixpkgs/pull/262506#pullrequestreview-1702897430

Reply:
> first building a combined name and then trying to parse the individual parts out again

This is not the case in this PR. If pname or version is passed in then they are used as is with no modification. Only if pname or version is missing are they extracted from name.

Reply:
> a better fix for this issue would be to adjust wrapType2 to take version + pname without name and _stop passing custom combined names_ everywhere.

I can move `name` into `args` as well, but [cases like this](https://github.com/NixOS/nixpkgs/blob/05f1030fa3aa75a7baf72bc016304ef1f647e7fc/pkgs/build-support/appimage/default.nix#L64C7-L64C7) still call for a combined `name`. I'll see what i can do

Reply:
I've got my changes cleaned up. There's still some bugs left to squish but just so that we don't duplicate work here: https://github.com/Atemu/nixpkgs/commit/258615c7d351e05b02dd2797779e493ee27fe34a

Reply:
And duplicate we did. I went in a direction where we don't use `null`, which greatly simplifies the logic